src/main/java/org/reactome/addlinks/fileprocessors/HPAFileProcessor.java

```diff
@@ -0,0 +1,53 @@
+package org.reactome.addlinks.fileprocessors;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class HPAFileProcessor extends FileProcessor<List<String>>
+{
+
+    public HPAFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public HPAFileProcessor()
+    {
+        super(null);
+    }
+
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        try {
+            String dirToHPAFiles = this.unzipFile(this.pathToFile);
+            String inputFilepath = dirToHPAFiles + "/" + this.pathToFile.getFileName().toString().replace(".zip", "");
+
+            for (String line : Files.readAllLines(Paths.get(inputFilepath)).stream().skip(1).collect((Collectors.toList()))) {
+                List<String> tabSplit = Arrays.asList(line.split("\t"));
+
+                String geneName = tabSplit.get(0);
+                String ensemblId = tabSplit.get(2);
+                String uniprotId = tabSplit.get(4);
+                String hpaUrlId = String.join("-", ensemblId, geneName);
+
+                if (!geneName.isEmpty() && !ensemblId.isEmpty() && !uniprotId.isEmpty()) {
+                    if (mappings.get(uniprotId) != null) {
+                        mappings.get(uniprotId).add(hpaUrlId);
+                    } else {
+                        mappings.put(uniprotId, new ArrayList<>(Arrays.asList(hpaUrlId)));
+
+                    }
```

Comment:
When adding a new entry to a list of values in a map, I usually prefer to use the `computeIfAbsent` method on the map.  That way no condition check is necessary, since if the key hasn't been used before an ArrayList will be created before adding the new value.

```
mappings.computeIfAbsent(uniprotId, k -> new ArrayList<>()).add(hpaUrlId);
```